src/image.cpp

```diff
@@ -184,7 +184,11 @@ bool Image::isBigEndianPlatform() {
 #endif
 }
 bool Image::isLittleEndianPlatform() {
+#ifdef __LITTLE_ENDIAN__
+  return true;
+#else
```

Comment:
Is this macro (`__LITTLE_ENDIAN__`) sometimes defined when `__BYTE_ORDER__` isn't? If so, maybe this logic should be moved to `isBigEndianPlatform`. I.e. add this to the beginning of `isBigEndianPlatform`:

```c
#if defined(__BIG_ENDIAN__)
  return true;
#elif defined(__LITTLE_ENDIAN__)
  return false;
#elif ...
```

Reply:
`__LITTLE_ENDIAN__` is our own macro defined in `config.h`.

Everything else comes from the compiler.

Reply:
> Is this macro (`__LITTLE_ENDIAN__`) sometimes defined when `__BYTE_ORDER__` isn't?

Potentially on WIN32 and MSVC (which AFAIK does not define `__BYTE_ORDER__`): https://github.com/Exiv2/exiv2/blob/50648dd4b0d2b61e3533278ca385255ee45f6eda/include/exiv2/config.h#L43

Reply:
@kevinbackhouse I'm now leveraging this as well.